docs/book/src/plugins/external-plugins.md

```diff
@@ -0,0 +1,155 @@
+# Extending Kubebuilder with external plugins
+
+## Overview
+
+You can extend the Kubebuilder functionality by introducing external plugin.
+
+An external plugin is an executable that provides the same kubebuilder plugin features. Since it is an executable, such customized plugin can be implemented in any language.
```

Comment:
Could you elaborate a bit more on what you mean by same kubebuilder plugin features? It might be nice to include those here.

Reply:
+1 to more elaboration. I would also recommend a bit of a rephrase here to be something more along the lines of:
```suggestion
An external plugin is an executable that implements an execution pattern that Kubebuilder knows how to interact with. Since an external plugin is an executable it can be implemented in any language. 
```